src/core/vdom/create-element.js

```diff
@@ -99,7 +99,7 @@ export function _createElement (
     // direct component options / constructor
     vnode = createComponent(tag, data, context, children)
   }
-  if (vnode !== undefined) {
+  if (vnode !== undefined && vnode !== null) {
```

Comment:
Sadly, flow does not support custom type predicate function.  

https://github.com/facebook/flow/issues/34

One workaround is to annotate `function isDef(obj): %checks {}`, using internal type check.